confgenerator/resourcedetector/gce_detector.go

```diff
@@ -162,7 +167,11 @@ func (r GCEResource) PrometheusStyleMetadata() map[string]string {
 
 	for k, v := range r.InterfaceIPv4 {
 		sanitizedKey := "interface_ipv4_nic" + strutil.SanitizeLabelName(k)
-		metaLabels[prefix+sanitizedKey] = v
+		// Once https://github.com/open-telemetry/opentelemetry-collector/issues/9204
+		// is fixed, this will no longer be needed.
+		sanitizedVal := strings.ReplaceAll(v, "${", "_{")
```

Comment:
Values of `InterfaceIPv4` should be plain IP addresses without user defined values, so may not need to be sanitized.

Reply:
Good point. Done.